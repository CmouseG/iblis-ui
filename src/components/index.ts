export { //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormData, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues, //
    WizardForm //
} from './form_components';
export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    HelperButton, HelperButtonProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, //
    IblisTextField, IblisTextFieldProps, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ProgressBar, ProgressBarProps, //
    UploadAvatar, UploadAvatarProps
} from './ui_components';
