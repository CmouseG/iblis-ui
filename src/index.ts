export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    ConfirmSignupForm, ConfirmSignupFormProps, ConfirmSignupFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    ForceChangePasswordForm, ForceChangePasswordFormProps, ForceChangePasswordFormValues, //
    HelperButton, HelperButtonProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, IblisTextField, IblisTextFieldProps, //
    LoginForm, LoginFormProps, LoginFormValues, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormData, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    PasswordConfirmForm, PasswordConfirmFormProps, PasswordConfirmFormValues, //
    ProgressBar, ProgressBarProps, //
    ResetPasswordForm, ResetPasswordFormProps, ResetPasswordFormValues, //
    SignupForm, SignupFormProps, SignupFormValues, //
    StripeForm, StripeFormProps, //
    UploadAvatar, UploadAvatarProps, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues //
} from './components';
export { //
    CardMenu, CardMenuLevelProps, CardMenuProps, //
    CardTitleLayout, CardTitleLayoutProps, //
    ContentWithTitle, ContentWithTitleProps, //
    ExternalFormLayout, ExternalFormLayoutProps, //
    MainLayout, MainLayoutProps
} from './layouts';
export { theme } from './theme';
// tslint:disable-next-line:no-import-side-effect
import 'iblis-font/index.css';
// tslint:disable-next-line:no-import-side-effect
import 'typeface-roboto/index.css';
