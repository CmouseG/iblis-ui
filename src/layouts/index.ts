export { CardMenu, CardMenuLevelProps, CardMenuProps } from './card_menu';
export { CardTitleLayout, CardTitleLayoutProps } from './card_title_layout';
export { ContentWithTitle, ContentWithTitleProps } from './content_with_title';
export { ExternalFormLayout, ExternalFormLayoutProps } from './external_form_layout';
export { MainLayout, MainLayoutProps } from './main_layout';
