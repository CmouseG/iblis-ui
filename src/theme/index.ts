export { theme } from './iblis_theme';
