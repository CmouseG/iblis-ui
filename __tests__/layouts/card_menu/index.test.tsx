import * as React from 'react';
import renderer from 'react-test-renderer';
import { CardMenu, CardMenuProps } from '../../../src';

it('renders correctly', () => {

    const menu = [
        {
            firstLevel: { name: 'Profile', action: () => undefined },
        },
        {
            firstLevel: { name: 'Security', action: () => undefined },
            sublevel: [
                { name: 'Your Details', action: () => undefined },
                { name: 'Change Password', action: () => undefined },
                { name: 'Delete Account', action: () => undefined },
            ],
        },
    ];

    const cardMenuProps: CardMenuProps = {
        menu: menu,
        firstLevelActive: 0,
        secondLevelActive: 0,
    };

    const tree = renderer
        .create(
            <CardMenu {...cardMenuProps}>
                {'Place Content Here'}
            </CardMenu>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
