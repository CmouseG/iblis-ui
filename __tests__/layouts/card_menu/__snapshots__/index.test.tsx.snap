// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5"
>
  <header
    className="MuiPaper-root-4 MuiPaper-elevation0-6 MuiAppBar-root-31 CardMenuBase-appBarMenuRoot-1 MuiAppBar-positionStatic-35"
  >
    <div
      className="MuiToolbar-root-39 CardMenuBase-toolBarMenuRoot-2 MuiToolbar-regular-41 MuiToolbar-gutters-40"
    >
      <div
        className="MuiTabs-root-43"
      >
        <div>
          <div
            style={
              Object {
                "height": 100,
                "msOverflowStyle": "scrollbar",
                "overflow": "scroll",
                "position": "absolute",
                "top": -10000,
                "width": 100,
              }
            }
          />
        </div>
        <div
          className="MuiTabs-flexContainer-44"
        >
          <div
            className="MuiTabs-scroller-46 MuiTabs-scrollable-48"
            onScroll={[Function]}
            role="tablist"
            style={
              Object {
                "marginBottom": NaN,
              }
            }
          >
            <div
              className="MuiTabs-flexContainer-44"
            >
              <button
                aria-selected={true}
                className="MuiButtonBase-root-64 MuiTab-root-52 MuiTab-textColorPrimary-55 MuiTab-selected-57"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                style={
                  Object {
                    "marginRight": 16,
                    "minWidth": 8,
                    "padding": 0,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                  }
                }
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiTab-wrapper-60"
                >
                  <span
                    className="MuiTab-labelContainer-61 CardMenuBase-tabLabelContainer-3"
                  >
                    <span
                      className="MuiTab-label-62"
                    >
                      Profile
                    </span>
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root-70"
                />
              </button>
              <button
                aria-selected={false}
                className="MuiButtonBase-root-64 MuiTab-root-52 MuiTab-textColorPrimary-55"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                style={
                  Object {
                    "marginRight": 16,
                    "minWidth": 8,
                    "padding": 0,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                  }
                }
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiTab-wrapper-60"
                >
                  <span
                    className="MuiTab-labelContainer-61 CardMenuBase-tabLabelContainer-3"
                  >
                    <span
                      className="MuiTab-label-62"
                    >
                      Security
                    </span>
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root-70"
                />
              </button>
            </div>
            <span
              className="TabIndicator-root-67 TabIndicator-colorPrimary-68 MuiTabs-indicator-51"
              style={
                Object {
                  "left": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    style={
      Object {
        "padding": 24,
      }
    }
  >
    Place Content Here
  </div>
</div>
`;
